lib/buffer.js

```diff
@@ -300,7 +299,7 @@ function fromString(string, encoding) {
   } else {
     length = byteLength(string, encoding, true);
     if (length === -1)
-      throw new TypeError('"encoding" must be a valid string encoding');
+      throw new errors.TypeError('ERR_INVALID_OPT_VALUE', 'encoding', encoding);
```

Comment:
but IMHO name should stay `encoding` like in the doc - https://nodejs.org/api/buffer.html#buffer_new_buffer_string_encoding